### DISCUSSION TITLE
TSAI for anomaly detection

### DISCUSSION THREAD

**Discussion by irazza** [timeseriesAI/tsai#736] [Q&A]
hello, I have been using tsai for a while now for timeseries classification problems, lately I have been working on anomaly detection. I was wondering if there was a way to use some architecture in the library to get a reconstruction error from the different signals and then use that as a threshold to see if a particular signal is an outlier or not. If there is no such possibility, I was wondering if there was a way to use one of the implemented architectures simply as a feature extractor and then from there try to calculate an anomaly score with statistical models. 
Do you know how to help me?
Thanks

**Comment by oguiza** [comment]
Hi @albertoazzari,
Anomaly detection is a complex topic since there are multiple types of anomalies.
One option is to use a forecasting model. You could set a threshold error utilizing the test data. And use that threshold to identify point anomalies. You could use any of the models in the library with this approach.
There are other models that use autoencoders to reconstruct the input and calculate the reconstruction error. But there's no autoencoder in `tsai` yet.
